sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsRelationshipTest.java

```diff
@@ -0,0 +1,231 @@
+            listIncomingRelationships.forEach(incomingRelationship -> incomingRelationshipsSourceIds.add(incomingRelationship.getSourceId()));
+            assertThat(incomingRelationshipsSourceIds)
+                .as("Floor has incoming relationships from room and hvac")
+                .containsExactlyInAnyOrder(roomTwinId, hvacTwinId);
+            logger.info("Retrieved incoming relationships for {}, found sources {}", floorTwinId, Arrays.toString(incomingRelationshipsSourceIds.toArray()));
+
+            // LIST relationships
+            List<String> relationshipsTargetIds = new ArrayList<>();
+            PagedIterable<BasicRelationship> listRelationships = client.listRelationships(floorTwinId, BasicRelationship.class);
+            listRelationships.forEach(basicRelationship -> relationshipsTargetIds.add(basicRelationship.getTargetId()));
+            assertThat(relationshipsTargetIds)
+                .as("Floor has a relationship to room and hvac")
+                .containsExactlyInAnyOrder(roomTwinId, hvacTwinId);
+            logger.info("Retrieved all relationships for {}, found targets {}", floorTwinId, Arrays.toString(relationshipsTargetIds.toArray()));
+
+            // LIST relationship by name
+            List<String> containedInRelationshipsTargetIds = new ArrayList<>();
+            PagedIterable<BasicRelationship> listContainedInRelationship = client.listRelationships(roomTwinId, CONTAINED_IN_RELATIONSHIP, BasicRelationship.class, Context.NONE);
+            listContainedInRelationship.forEach(basicRelationship -> {
+                containedInRelationshipsTargetIds.add(basicRelationship.getTargetId());
+                logger.info("Retrieved relationship {} for twin {}", basicRelationship.getId(), roomTwinId);
+            });
+            assertThat(containedInRelationshipsTargetIds.size())
+                .as("Room has only one containedIn relationship to floor")
+                .isEqualTo(1);
+
+            // DELETE the created relationships
+            client.deleteRelationship(floorTwinId, FLOOR_CONTAINS_ROOM_RELATIONSHIP_ID);
+            logger.info("Deleted relationship {} for twin {}", FLOOR_CONTAINS_ROOM_RELATIONSHIP_ID, floorTwinId);
+
+            client.deleteRelationship(roomTwinId, ROOM_CONTAINED_IN_FLOOR_RELATIONSHIP_ID);
+            logger.info("Deleted relationship {} for twin {}", ROOM_CONTAINED_IN_FLOOR_RELATIONSHIP_ID, roomTwinId);
+
+            client.deleteRelationship(floorTwinId, FLOOR_COOLED_BY_HVAC_RELATIONSHIP_ID);
+            logger.info("Deleted relationship {} for twin {}", FLOOR_COOLED_BY_HVAC_RELATIONSHIP_ID, floorTwinId);
+
+            client.deleteRelationship(hvacTwinId, HVAC_COOLS_FLOOR_RELATIONSHIP_ID);
+            logger.info("Deleted relationship {} for twin {}", HVAC_COOLS_FLOOR_RELATIONSHIP_ID, hvacTwinId);
+
+            // GET a relationship which doesn't exist - should return status code 404 (Not Found).
```

Comment:
Could this be split out into a separate test? This test is already quite long

Reply:
I can split it into a separate test if you feel strongly about it; I added it in here because of the convinience (the models are twins were already created).